### PR TITLE
Move DF templates into a Data Foundation template package

### PR DESCRIPTION
**Pull request by bindrad**
Separating the Data Foundation related templates to the templates package

### CODE REVIEW

File: controllers/datafoundation.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ import (
 	promv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
 	v1alpha1 "github.com/red-hat-storage/managed-fusion-agent/api/v1alpha1"
 	"github.com/red-hat-storage/managed-fusion-agent/datafoundation"
+	dftemplates "github.com/red-hat-storage/managed-fusion-agent/datafoundation/templates"
```
Comment (nb-ohad) [inline controllers/datafoundation.go:28]: In this context, the data foundation templates are the main source for templates for this code (not a secondary one). I will suggest to just call it `templates`
 ```suggestion
	"github.com/red-hat-storage/managed-fusion-agent/datafoundation/templates"
```

File: datafoundation/templates/storagecluster.go [hunk 2]
```diff
@@ -10,11 +10,12 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package datafoundation
+package templates
 
 import (
 	"fmt"
 
+	"github.com/red-hat-storage/managed-fusion-agent/datafoundation"
```
Comment (nb-ohad) [inline datafoundation/templates/storagecluster.go:18]:
```suggestion
	df "github.com/red-hat-storage/managed-fusion-agent/datafoundation"
```

**Review by nb-ohad — CHANGES_REQUESTED** [review]

File: datafoundation/templates/storagecluster.go (hunk 2 above)
Comment (bindrad) [reply datafoundation/templates/storagecluster.go]: Updated

File: controllers/datafoundation.go (hunk 1 above)
Comment (bindrad) [reply controllers/datafoundation.go]: Updated

**Pull request merged**